src/plugins/fs.c

```diff
@@ -47,6 +47,29 @@
  * A plugin for operations with file systems
  */
 
+const BDFSInfo bd_fs_info[] = {
+    {"xfs", "xfs_db", "xfs_repair", "xfs_growfs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW},
+    {"ext2", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"ext3", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"ext4", "e2fsck", "e2fsck", "resize2fs", BD_FS_ONLINE_GROW | BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK},
+    {"vfat", "fsck.vfat", "fsck.vfat", "", BD_FS_OFFLINE_GROW | BD_FS_OFFLINE_SHRINK}
+};
+
+const BDFSInfo *
+bd_fs_get_info (const gchar *type)
+{
+    guint n;
+
+    g_return_val_if_fail (type != NULL, NULL);
+
+    for (n = 0; n < sizeof(bd_fs_info)/sizeof(BDFSInfo); n++) {
```

Comment:
Also I think it would be better and more reliable to terminate the ``bd_fs_info`` array with ``NULL`` and make use of it in this cycle. The compiler can do some alignments possibly making the ``sizeof(bd_fs_info)/sizeof(BDFSInfo)`` calculation wrong.